// eslint-plugin-suitescript/lib/index.js
/**
 * @fileoverview Adds an ESLint Environment that describes all global functions and objects defined by the NetSuite SuiteScript API
 * @author Eric Grubaugh
 * @copyright 2016 Eric Grubaugh. All rights reserved.
 * See LICENSE file in root directory for full license.
 */
"use strict";

//------------------------------------------------------------------------------
// Requirements
//------------------------------------------------------------------------------
var requireIndex = require("requireindex");

//------------------------------------------------------------------------------
// Plugin Definition
//------------------------------------------------------------------------------
module.exports = {
    environments: {
        suitescript1: {
            globals: {
                nlapiCopyRecord : false,
				nlapiLoadRecord : false,
				nlapiCreateRecord : false,
				nlapiSubmitRecord : false,
				nlapiDeleteRecord : false,
				nlapiSearchRecord : false,
				nlapiSearchGlobal : false,
				nlapiSearchDuplicate : false,
				nlapiTransformRecord : false,
				nlapiVoidTransaction  : false,
				nlapiLookupField : false,
				nlapiSubmitField : false,
				nlapiAttachRecord : false,
				nlapiDetachRecord : false,
				nlapiResolveURL : false,
				nlapiSetRedirectURL : false,
				nlapiRequestURL : false,
				nlapiGetContext : false,
				nlapiGetUser : false,
				nlapiGetRole : false,
				nlapiGetDepartment : false,
				nlapiGetLocation : false,
				nlapiGetSubsidiary : false,
				nlapiGetRecordType : false,
				nlapiGetRecordId : false,
				nlapiSendEmail : false,
				nlapiSendCampaignEmail : false,
				nlapiSendFax : false,
				nlapiGetField : false,
				nlapiGetMatrixField : false,
				nlapiGetLineItemField : false,
				nlapiGetLineItemMatrixField : false,
				nlapiGetFieldValue : false,
				nlapiSetFieldValue : false,
				nlapiGetFieldText : false,
				nlapiSetFieldText : false,
				nlapiGetFieldValues : false,
				nlapiSetFieldValues : false,
				nlapiGetFieldTexts : false,
				nlapiSetFieldTexts : false,
				nlapiGetMatrixValue : false,
				nlapiSetMatrixValue : false,
				nlapiGetCurrentLineItemMatrixValue : false,
				nlapiSetCurrentLineItemMatrixValue : false,
				nlapiGetLineItemMatrixValue : false,
				nlapiGetLineItemValue : false,
				nlapiGetLineItemDateTimeValue : false,
				nlapiGetLineItemDateTimeValue : false,
				nlapiSetLineItemValue : false,
				nlapiSetLineItemDateTimeValue : false,
				nlapiSetLineItemDateTimeValue : false,
				nlapiGetLineItemText : false,
				nlapiFindLineItemValue : false,
				nlapiFindLineItemMatrixValue : false,
				nlapiGetMatrixCount : false,
				nlapiGetLineItemCount : false,
				nlapiInsertLineItem : false,
				nlapiRemoveLineItem : false,
				nlapiSetCurrentLineItemValue : false,
				nlapiSetCurrentLineItemDateTimeValue : false,
				nlapiSetCurrentLineItemDateTimeValue : false,
				nlapiSetCurrentLineItemText : false,
				nlapiGetCurrentLineItemValue : false,
				nlapiGetCurrentLineItemDateTimeValue : false,
				nlapiGetCurrentLineItemDateTimeValue : false,
				nlapiGetCurrentLineItemText : false,
				nlapiGetCurrentLineItemIndex : false,
				nlapiSetLineItemDisabled : false,
				nlapiGetFieldMandatory : false,
				nlapiGetLineItemMandatory : false,
				nlapiSetFieldMandatory : false,
				nlapiSetLineItemMandatory : false,
				nlapiSelectLineItem : false,
				nlapiCommitLineItem : false,
				nlapiCancelLineItem : false,
				nlapiSelectNewLineItem : false,
				nlapiRefreshLineItems : false,
				nlapiInsertSelectOption : false,
				nlapiRemoveSelectOption : false,
				nlapiInsertLineItemOption : false,
				nlapiRemoveLineItemOption : false,
				nlapiIsLineItemChanged : false,
				nlapiGetNewRecord : false,
				nlapiGetOldRecord : false,
				nlapiCreateError : false,
				nlapiCreateForm : false,
				nlapiCreateList : false,
				nlapiCreateAssistant : false,
				nlapiLoadFile : false,
				nlapiSubmitFile : false,
				nlapiDeleteFile : false,
				nlapiCreateFile : false,
				nlapiMergeRecord : false,
				nlapiPrintRecord : false,
				nlapiXMLToPDF : false,
				nlapiCreateTemplateRenderer : false,
				nlapiCreateEmailMerger : false,
				nlapiLogExecution : false,
				nlapiScheduleScript : false,
				nlapiOutboundSSO : false,
				nlapiLoadConfiguration : false,
				nlapiSubmitConfiguration : false,
				nlapiStringToDate : false,
				nlapiDateToString : false,
				nlapiAddDays : false,
				nlapiAddMonths : false,
				nlapiFormatCurrency : false,
				nlapiEncrypt : false,
				nlapiEscapeXML : false,
				nlapiStringToXML : false,
				nlapiXMLToString : false,
				nlapiValidateXML : false,
				nlapiSelectValue : false,
				nlapiSelectValues : false,
				nlapiSelectNode : false,
				nlapiSelectNodes : false,
				nlapiExchangeRate : false,
				nlapiInitiateWorkflow : false,
				nlapiInitiateWorkflowAsync : false,
				nlapiTriggerWorkflow : false,
				nlapiCreateCurrentLineSubrecord : false,
				nlapiEditCurrentLineItemSubrecord : false,
				nlapiRemoveCurrentLineItemSubrecord : false,
				nlapiViewCurrentLineItemSubrecord : false,
				nlapiViewLineItemSubrecord : false,
				nlapiGetCache : false,
				createSubrecord : false,
				editSubrecord : false,
				removeSubrecord : false,
				viewSubrecord : false,
				nlobjSubrecord : false,
				nlobjRecord : false,
				nlobjConfiguration : false,
				nlobjFile : false,
				nlobjSearchFilter : false,
				nlobjSearchColumn : false,
				nlobjSearchResult : false,
				nlobjContext : false,
				nlobjError : false,
				nlobjServerResponse : false,
				nlobjTemplateRenderer : false,
				nlobjEmailMerger : false,
				nlobjResponse : false,
				nlobjRequest : false,
				nlobjPortlet : false,
				nlobjList : false,
				nlobjForm : false,
				nlobjAssistant : false,
				nlobjField : false,
				nlobjSubList : false,
				nlobjColumn : false,
				nlobjTab : false,
				nlobjAssistantStep : false,
				nlobjFieldGroup : false,
				nlobjButton : false,
				nlobjSelectOption : false,
				nlapiGetLogin : false,
				nlobjLogin : false,
				nlapiGetJobManager : false,
				nlobjJobManager : false,
				nlobjDuplicateJobRequest : false,
				nlobjFuture : false,
				nlobjCache : false
            }
        },
		suitescript2: {
            globals: {
				log : false,
				util : false
			}
		}
    }
};
